src/tester.rs

```diff
@@ -907,9 +907,13 @@ fn run_cmd(
                 }
                 if stdin_off == stdin_str.len() {
                     stdin_finished = true;
-                    // This is a bit icky, because the `stdin` variable will later close stdin when the
-                    // variable is dropped. However, some programs expect stdin to be closed before
-                    // they'll continue, so this is the least worst option.
+                    // If we don't close the child's stdin now, the child process might hang,
+                    // waiting forever for input that will never come. However, if we're not
+                    // careful, we'll close the underlying file handle twice: once here and once
+                    // when Rust drops the `stdin` variable. Since file handles can be reused that
+                    // means we could close a file handle that's now being used elsewhere. Thus
+                    // because we close the child's stdin here, we need to make sure we `forget`
+                    // the `stdin` variable at every return point from this function.
```

Comment:
If you explicitly drop stdin here (maybe not possible?) instead of closing it, would we then need the forgets?

Reply:
It can't be dropped here, as Rust won't be able to prove that it's not used on the next iteration of the loop.